internal/middleware/account.go

```diff
@@ -0,0 +1,34 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+)
+
+func AccountMiddleware(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		rhId := ctxval.GetIdentity(r.Context())
+		logger := ctxval.GetLogger(r.Context())
+		accDao, err := dao.GetAccountDao(r.Context())
+		if err != nil {
+			logger.Error().Err(err).Msg("Failed to initialize connection to fetch Account info")
+			panic(err)
+		}
+		acc, err := accDao.GetOrCreateByIdentity(r.Context(), rhId.Identity.OrgID, rhId.Identity.AccountNumber)
+		if err != nil {
+			logger.Warn().Err(err).Msg("Failed to fetch account by identity")
+			// TODO do something about it, or fail once we have create implemented
```

Comment:
Please error out with 500 here, we must not continue at this point it's a fatal error and security related, so no TODOs please anymore :)

Reply:
You know it will break the app now, right? So we need to wait for Create in that case :)

Reply:
I've improved the TODO message, and put a commentable implementation how I'd return 500 there.
Tho if you insist on this comment, we need to have `account.Create` in place prior merging this.
As also explained in https://github.com/RHEnVision/provisioning-backend/pull/90#discussion_r923295961